orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/pipeline/WebhookStage.java

```diff
@@ -0,0 +1,146 @@
+package com.netflix.spinnaker.orca.webhook.pipeline;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.netflix.spinnaker.kork.exceptions.UserException;
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder;
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageGraphBuilder;
+import com.netflix.spinnaker.orca.api.pipeline.graph.TaskNode;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.pipeline.tasks.WaitTask;
+import com.netflix.spinnaker.orca.pipeline.tasks.artifacts.BindProducedArtifactsTask;
+import com.netflix.spinnaker.orca.webhook.config.WebhookProperties;
+import com.netflix.spinnaker.orca.webhook.tasks.CreateWebhookTask;
+import com.netflix.spinnaker.orca.webhook.tasks.MonitorWebhookTask;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class WebhookStage implements StageDefinitionBuilder {
+
+  private final MonitorWebhookTask monitorWebhookTask;
+
+  @Autowired
+  public WebhookStage(MonitorWebhookTask monitorWebhookTask) {
+    this.monitorWebhookTask = monitorWebhookTask;
+  }
+
+  @Override
+  public void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder builder) {
+    StageData stageData = stage.mapTo(StageData.class);
+
+    if (stageData.monitorOnly && !stageData.waitForCompletion) {
+      throw new UserException(
+          "Can't specify monitorOnly = true and waitForCompletion = false at the same time");
+    }
+
+    if (!stageData.monitorOnly) {
+      builder.withTask("createWebhook", CreateWebhookTask.class);
+    }
+    if (stageData.waitForCompletion) {
+      if (stageData.waitBeforeMonitor > 0) {
+        stage.getContext().putIfAbsent("waitTime", stageData.waitBeforeMonitor);
+        builder.withTask("waitBeforeMonitorWebhook", WaitTask.class);
+      }
+
+      builder.withTask("monitorWebhook", MonitorWebhookTask.class);
+    }
+    if (stage.getContext().containsKey("expectedArtifacts")) {
+      builder.withTask(BindProducedArtifactsTask.TASK_NAME, BindProducedArtifactsTask.class);
+    }
+  }
+
+  @Override
+  public void onFailureStages(@Nonnull StageExecution stage, @Nonnull StageGraphBuilder graph) {
+    monitorWebhookTask.onCancel(stage);
+  }
+
+  @Data
+  public static class StageData {
+    // Inputs for webhook
+    public String url;
+    public Object payload;
+    public Map<String, Object> customHeaders;
+    public List<Integer> failFastStatusCodes;
+    public Boolean waitForCompletion;
+    public WebhookProperties.StatusUrlResolution statusUrlResolution;
+    public String statusUrlJsonPath;
+    public boolean monitorOnly;
+
+    @JsonFormat(with = {JsonFormat.Feature.ACCEPT_CASE_INSENSITIVE_PROPERTIES})
```

Comment:
I'm a bit rusty on my lombok but it is worth double checking that these annotations end up applying with the `@Data` class generating getters/setters

Reply:
I added a test case to ensure it is working as expected